test/testenv/testenv.go

```diff
@@ -0,0 +1,83 @@
+package testenv
+
+import (
+	"io"
+	"net"
+
+	"google.golang.org/grpc"
+	mixerpb "istio.io/api/mixer/v1"
+	"istio.io/mixer/cmd/server/cmd"
+	"istio.io/mixer/cmd/shared"
+	"istio.io/mixer/pkg/handler"
+	"istio.io/mixer/pkg/template"
+)
+
+// TestEnv interface
+type TestEnv interface {
+	io.Closer
+	CreateMixerClient() (mixerpb.MixerClient, *grpc.ClientConn, error)
+}
+
+type testEnv struct {
+	addr         string
+	mixerContext *cmd.ServerContext
+	shutdown     chan struct{}
+}
+
+// Args includes the required args to initialize Mixer server.
+type Args struct {
+	MixerServerAddr               string
+	ConfigStoreURL                string
+	ConfigStore2URL               string
+	ConfigDefaultNamespace        string
+	ConfigIdentityAttributeDomain string
+}
+
+// NewEnv creates a TestEnv instance.
+func NewEnv(args *Args, info map[string]template.Info, adapters []handler.InfoFn) (TestEnv, error) {
```

Comment:
Just `New`, we're already in package `testenv` so at usage sites this will read `testenv.New` rather than `testenv.NewEnv`, which stutters.

Reply:
That would prevent us from adding other New

Reply:
Sure - when we add another thing to this package that needs a `New*` method, rename it. For now you have a single public _thing_ that this package does, so call this method New so it reads naturally.